### PR TITLE
WIP Refactoring Normalizer to break out hard coded XPath expressions.

### PR DESCRIPTION
**Pull request by tingulfsen**
@drh-stanford Please comment and review.

### CODE REVIEW

File: lib/modsulator/normalizer.rb [hunk 1]
```diff
@@ -165,12 +174,12 @@ def trim_text(node)
...
   # @return [Void]                    The given document is modified in place.
-  def clean_date_attributes(root)
-    
+  def clean_date_attributes(nodes)
+
     # Find all the <dateCreated> and <dateIssued> elements that are NOT immediately followed by another element with the same name
-    root.xpath('//mods:originInfo/mods:dateCreated[1][not(following-sibling::*[1][self::mods:dateCreated])] | //mods:originInfo/mods:dateIssued[1][not(following-sibling::*[1][self::mods:dateIssued])]', 'mods' => 'http://www.loc.gov/mods/v3').each do |current_element|
+    nodes.each do |current_element|
       attributes = current_element.attributes
       if(attributes.key?('point'))
```
Comment (drh-stanford) [inline lib/modsulator/normalizer.rb]: This bit is still tied to the MODS schema I think? Not sure what the `point` attribute is.

**Comment by tingulfsen** [comment]
@drh-stanford correct, that method is kind of tied to the MODS schema in the sense that it requires the element to have a 'point' attribute to act on it. Is that a dealbreaker for you? We could rename the method or create another method for you.

**Comment by LynnMcRae** [comment]
What is distinct about that "point" attribute in MODS dates (used for ranges, point="start" and point="end") is that it is specific to solving a problem in consuming the replayable spreadsheet template, not likely a normalization that would be needed for any other real-world situation. Does that suggest any alternative?

**Comment by drh-stanford** [comment]
I think it's fine. I won't need to call that function anyhow, but it could be renamed in the future.

**Pull request merged**